rclcpp/include/rclcpp/strategies/allocator_memory_strategy.hpp

```diff
@@ -164,6 +164,14 @@ class AllocatorMemoryStrategy : public memory_strategy::MemoryStrategy
         if (!group || !group->can_be_taken_from().load()) {
           continue;
         }
+        for (auto & weak_publisher : group->get_publisher_ptrs()) {
```

Comment:
This whole loop is technically unsafe due to the mutex call out above.

Reply:
Will address in a separate PR.